mlir/lib/Dialect/MIOpen/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -378,22 +220,29 @@ struct BlockwiseGemmRewritePattern : public OpRewritePattern<BlockwiseGemmOp> {
           lab.create<MulIOp>(loc, iva, MPerThreadSubCConstantOp)};
     }
 
-    // Emit threadwise_copy.
-    auto threadwiseCopyAMatrixOp = lab.create<ThreadwiseCopyOp>(
-        loc, op.matrixA(), threadAAllocOp,
-        b.getArrayAttr({transformsA, emptyArr}), noPadding, noOobDims,
-        noGlobals, matrixAThreadwiseCopySourceCoords,
-        matrixAThreadwiseCopyDestCoords);
-    affixThreadwiseCopyAttributes(threadwiseCopyAMatrixOp, op, b,
-                                  /*isMatrixA=*/true);
+    auto copyALoop = lab.create<TransformingForOp>(
+        loc,
+        ArrayRef<ValueRange>{matrixAThreadwiseCopySourceCoords,
+                             matrixAThreadwiseCopyDestCoords},
+        ArrayRef<Attribute>{transformsA, emptyArr},
+        ArrayRef<int64_t>{1, KPerThread, MPerThreadSubC},
+        /*forceUnroll=*/true, /*indexDiffs=*/false);
```

Comment:
Any particular reason why index diffs is not in use here?

Reply:
Because they weren't in the original